templates/deno/src/services/service.ts.twig

```diff
@@ -148,7 +149,20 @@ export class {{ service.name | caseUcfirst }} extends Service {
 
                     for (let byteIndex = 0; byteIndex < Client.DENO_READ_CHUNK_SIZE; byteIndex++) {
                         totalBuffer[(blockIndex * Client.DENO_READ_CHUNK_SIZE) + byteIndex] = buf[byteIndex];
+
+                        if(buf[byteIndex] !== 0) {
+                            lastBufferIndex = (blockIndex * Client.DENO_READ_CHUNK_SIZE) + byteIndex;
+                        }
+                    }
+                }
+
+                // Shrink empty bytes
+                if(lastBufferIndex !== -1) {
+                    const newTotalBuffer = new Uint8Array(lastBufferIndex + 1);
+                    for(let index = 0; index <= lastBufferIndex; index++) {
+                        newTotalBuffer[index] = totalBuffer[index];
```

Comment:
Could we use an `ArrayBuffer` and create the final `Uint8Array` using the `new Uint8Array(buffer, byteOffset, length)` constructor to avoid copying the bytes here?

Reply:
It took quite a lot of try&error to get the code working and I am honestly scared to touch it. I am making a note to myself to revisit it later, but to make sure this makes it into the next release, I would prefer to skip this improvement.

Reply:
Let's keep it for now - but improve it in the future. Considering the release date 👍🏻